test/studies/lcals/LCALSMain.chpl

```diff
@@ -0,0 +1,1203 @@
+      var stat = suite_run_info.getLoopStats(variant_name)[iloop];
+      if loop_names[iloop].length != 0 && stat.loop_is_run {
+        writer.write(loop_names[iloop]);
+        for ilen in 0..#stat.loop_length.size() {
+          writer.writef("%s%r", sepchr, stat.mean[ilen]);
+        }
+        writer.writeln();
+      }
+    }
+    writer.close();
+    outfile.close();
+  }
+
+  proc writeRelativeTimeReport(variant_name: string, output_dirname: string) {
+    var rept_fname = output_dirname + "/" + variant_name + "-reltime.txt";
+    var outfile = open(rept_fname, iomode.cw);
+    var writer = outfile.writer();
+    writer.writeln("\n writeRelativeTimeReport...   ");
+
+    var suite_run_info = getLoopSuiteRunInfo();
+    var loop_names = suite_run_info.loop_names;
+    var len_names = suite_run_info.loop_length_names;
+
+    const sepchr = " , ";
+    const prec = 6;
+
+    writer.write(variant_name, " Relative Run Times ");
+    for i in 0..#len_names.size() {
+      writer.write(sepchr);
+    }
+    writer.writeln();
+
+    for i in 0..#len_names.size() {
+      writer.write(sepchr, len_names[i]);
+    }
+    writer.writeln();
+
+    for iloop in 0..#loop_names.size() {
+      var stat = suite_run_info.getLoopStats(variant_name)[iloop];
+      if loop_names[iloop].length != 0 && stat.loop_is_run {
+        writer.write(loop_names[iloop]);
+        for ilen in 0..#stat.loop_length.size() {
+          writer.write(sepchr, stat.meanrel2ref[ilen]);
+        }
+        writer.writeln();
+      }
+    }
+    writer.close();
+    outfile.close();
+  }
+
+  proc writeTimingSummaryReport(run_loop_variants: vector(string), outchannel) {
+    var suite_run_info = getLoopSuiteRunInfo();
+    const nvariants = run_loop_variants.size();
+    const ref_variant = run_loop_variants[0];
+    var loop_names = suite_run_info.loop_names;
+
+    const equal_line = "===========================================================================================================\n";
+    const dash_line = "------------------------------------------------------------------------------------------------------------\n";
+    const dash_line_part = "-------------------------------------------------------\n";
+    const dot_line_part = "............................................\n";
+    var len_id = new vector(string);
+    len_id.resize(suite_run_info.loop_length_names.size());
+    for ilen in 0..#len_id.size() {
+      len_id[ilen] = suite_run_info.loop_length_names[ilen].substring(1);
+    }
+
+    var ver_info = buildVersionInfo();
+
+    outchannel.write("\n\n\n");
+    outchannel.write(equal_line);
+    outchannel.write(equal_line);
+
+    outchannel.writeln("LCALS compilation summary: ");
+    outchannel.writeln(ver_info);
+
+    outchannel.write("\n\n");
+    outchannel.write(equal_line);
+    outchannel.write(equal_line);
+
+    outchannel.writeln("LCALS run summary: ");
+    outchannel.writeln("sizeof(Real_type) = ", 8 /*sizeof(Real_type)*/);
```

Comment:
No.  Changed it to numBytes(real).  So far I've hard coded the types to use Chapel 'real's, while the reference used macros everywhere to allow it to switch between 32 and 64 bit float.  I don't currently have a type defined called Real_type since I forced it to 64 bit reals.  Allowing a trivial switch between 32 and 64 bit would be a nice improvement as I get into the 'cleanup and chapel-ize' steps.